### ISSUE TITLE
Unicode fallout

### ISSUE THREAD

**Issue by Simn** [HaxeFoundation/haxe#7383]
## Lua

The `Sha224` and `Sha256` tests are still ignored.

## Python

 <strike>I had to ignore one `substr` test here: https://github.com/HaxeFoundation/haxe/commit/d6ff859d85747a29cc9ad647f17e999d782395a7#diff-be86ed1cdbb99dd94a0a52804ac8d953R160

That's probably a general problem unrelated to Unicode.</strike>

Like with lua, the `Sha224` and `Sha256` tests are still ignored.

## HL

<strike>
`src/unitstd/Unicode.unit.hx:107: éあ������＀� should be éあ������`
</strike>

## Eval

<strike>I kind of forgot to actually check this `encoding` parameter anywhere... which also means that there are no tests for the `RawNative` version.</strike>

**Comment by frabbit** [comment]
the substr issue is fixed. How should str2blks actually work, should it yield the same result on all platforms? (conversion to a common encoding first?)

**Comment by Simn** [comment]
The HL problem seems to be contagious:

```
src/unitstd/Unicode.unit.hx:107: éあ������︀̝ should be éあ������
src/unitstd/Unicode.unit.hx:130: éあ������︀̝ should be éあ������
```

**Comment by ncannasse** [comment]
HL has been fixed in f3dc67e14f69428ed857cc02390f8881de3cc0fb

**Comment by ncannasse** [comment]
I think we're good there for now.
